isites_migration/utils.py

```diff
@@ -0,0 +1,348 @@
+        try:
+            os.makedirs(os.path.join(settings.EXPORT_DIR, settings.CANVAS_IMPORT_FOLDER_PREFIX + keyword))
+        except os.error:
+            pass
+
+        site = Site.objects.get(keyword=keyword)
+
+        _export_readme(keyword)
+
+        query_set = Topic.objects.filter(site=site).exclude(
+            Q(tool_id__in=settings.EXPORT_FILES_EXCLUDED_TOOL_IDS) |
+            Q(title__in=settings.EXPORT_FILES_EXCLUDED_TOPIC_TITLES)
+        ).only(
+            'topic_id', 'title'
+        )
+        logger.info('Attempting to export files for %d topics', query_set.count())
+        for topic in query_set:
+            if topic.title:
+                topic_title = topic.title.strip().replace(' ', '_')
+            else:
+                topic_title = 'no_title_%s' % topic.topic_id
+
+            file_repository_id = "icb.topic%s.files" % topic.topic_id
+            try:
+                file_repository = FileRepository.objects.select_related('storage_node').only(
+                    'file_repository_id', 'storage_node'
+                ).get(file_repository_id=file_repository_id)
+                _export_file_repository(file_repository, keyword, topic_title)
+            except FileRepository.DoesNotExist:
+                logger.info("FileRepository does not exist for %s", file_repository_id)
+                continue
+
+            _export_topic_text(topic, keyword, topic_title)
+
+        zip_path_index = len(settings.EXPORT_DIR) + 1
+        keyword_export_path = os.path.join(settings.EXPORT_DIR, settings.CANVAS_IMPORT_FOLDER_PREFIX + keyword)
+        zip_filename = os.path.join(settings.EXPORT_DIR, "%s%s.zip" % (settings.CANVAS_IMPORT_FOLDER_PREFIX, keyword))
+        z_file = zipfile.ZipFile(zip_filename, 'w')
+        for root, dirs, files in os.walk(keyword_export_path):
+            for file in files:
+                file_path = os.path.join(root, file)
+                z_file.write(file_path, file_path[zip_path_index:])
+        z_file.close()
+        shutil.rmtree(keyword_export_path)
+
+        export_key = Key(s3_bucket)
+        export_key.key = "%s.zip" % keyword
+        export_key.set_metadata('Content-Type', 'application/zip')
+        export_key.set_contents_from_filename(zip_filename)
+        logger.info("Uploaded file export for keyword %s to S3 Key %s", keyword, export_key.key)
+
+        os.remove(zip_filename)
+
+        logger.info("Finished exporting files for keyword %s to S3 bucket %s", keyword, s3_bucket.name)
+    except Exception:
+        message = "Failed to complete file export for keyword %s", keyword
+        logger.exception(message)
+        raise RuntimeError(message)
+
+
+def import_files(keyword, canvas_course_id):
+    try:
+        progress_url = create_canvas_content_migration(keyword, canvas_course_id)
+
+        workflow_state = 'processing'
+        while workflow_state not in ['completed', 'failed']:
```

Comment:
That's fine, we can always fire up more workers and kill off the zombies. Let's see if we run into any problems and make things more robust incrementally.